proto/compiler.bzl

```diff
@@ -31,6 +35,15 @@ load(
     "go_reset_target",
 )
 
+# This is actually a misuse of Proto toolchains: The proper way to use `protoc` would be to go
+# through a Go-specific `proto_lang_toolchain` and use the methods on `proto_common` to interact
+# with `protoc`. Since rules_go has a very bespoke setup with customizable compilers and the need
+# to apply reset transitions in case `protoc` *is* built from source, this would require major
+# changes.
+# TODO: Revisit this after --incompatible_enable_proto_toolchain_resolution has been enabled by
+#  default.
+_PROTO_TOOLCHAIN_TYPE = "@rules_proto//proto:toolchain_type"
```

Comment:
Neither https://bazel.build/rules/lib/toplevel/proto_common
or https://github.com/search?q=repo%3Abazelbuild%2Frules_proto%20%2Fproto_common%2F&type=code provide me with any inspiration.

Took me a while to realize that `proto_common` here is not a starlark struct like `cc_common` or `java_common`, but is a file located in `@rules_proto//proto:proto_common.bzl`. This inconsistency is quite rough... I wonder if we could do `s/proto_toolchains/proto_common/` here (or in rules_proto).

Reply:
`proto_toolchains` is a helper struct that just happens to be colocated with `proto_common`. `proto_common` is the equivalent of `cc_common`, with the difference being that it is already loaded from its canonical post-Starlarkification location.

I know there is a lot going on here, so please let me know if I should add any comments.